### PR TITLE
rename references from discordapp.com to discord.com

### PR DESCRIPTION
**Pull request by areyouokevan**
### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/webhook.py [hunk 1]
```diff
@@ -504,7 +504,7 @@ def from_url(cls, url, *, adapter):
             The URL is invalid.
         """
 
-        m = re.search(r'discordapp.com/api/webhooks/(?P<id>[0-9]{17,21})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})', url)
+        m = re.search(r'discord.com/api/webhooks/(?P<id>[0-9]{17,21})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})', url)
```
Comment (iomintz) [inline discord/webhook.py:507]: this should be changed to make `app` optional, not to require the new URL.

Comment (areyouokevan) [reply discord/webhook.py:507]: As per the announcement channel in Discord API
> We currently plan to start requiring the new domain later this year for third party developers on November 7th, 2020. We will be sending out more formal notices (system DMs and account emails) to announce this breaking change in the coming weeks.

Comment (iomintz) [reply discord/webhook.py:507]: Just because Discord is gonna be breaking old URLs doesn't mean we have to do the same.

Comment (jacc) [reply discord/webhook.py:507]: Make it `app` optional until November 7th, 2020, which at that point force users to use `discord.com` and not `discordapp.com`.

Comment (diceroll123) [reply discord/webhook.py:507]: To put a time on `app` being there is to create a breaking change. It does not matter when. You have to consider people who set-and-forget bots with perfectly fine, yet old webhooks in their configs.

Comment (C0braD3v) [reply discord/webhook.py:507]: All webhooks that are `discordapp.com` also work on the new `discord.com` domain, so any webhook ID/Token that is a valid webhook at `discordapp.com` will also be valid at `discord.com`

Comment (diceroll123) [reply discord/webhook.py:507]: Not if they're not parsed here, they won't.

Comment (C0braD3v) [reply discord/webhook.py:507]: In this case, could you not just add regex to check for either one?
```py
m = re.search(r'(discord|discordapp).com/api/webhooks/(?P<id>[0-9]{17,21})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})', url)```

Comment (Rapptz) [inline discord/webhook.py:507]:
```suggestion
        m = re.search(r'discord(?:app)?.com/api/webhooks/(?P<id>[0-9]{17,21})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})', url)
```

**Comment by DleanJeans** [comment]
[MessageConverter](https://github.com/Rapptz/discord.py/blob/master/discord/ext/commands/converter.py#L197) is currently not working with `jump_url`. Removing `app` fixes it.

**Comment by Rapptz** [comment]
Since this PR wasn't worked on I did it myself (along with the message converter issue). Thanks anyway.

**Pull request closed without merging**